torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -327,88 +407,90 @@ static c10::optional<MatchedSchema> tryMatchSchema(
+      if (arg.name() == "self" && self) {
+        actual_named_value = self;
+        self = c10::nullopt;
+      } else if (!arg.kwarg_only() && used_args < args.size()) {
+        // Try to convert all the remaining non-kwarg arguments (used_args) to a
+        // list. Allow zeros(IntArrayRef sizes) to work with zeros(1, 2) or
+        // zeros(1)
+        if (allow_conversions && varargsCanBeUsedAsList(schema, schema_i, arg)) {
+          auto value = args[used_args].value(graph);
+          const auto& actual_type = value->type();
+          // The actual cannot already be a list
+          if (actual_type->kind() != TypeKind::ListType &&
+              !convertibleToList(actual_type, unwrapOptional(arg.type()))) {
+            auto formal_type = unwrapOptional(arg.type())
+                                  ->expectRef<ListType>()
+                                  .getElementType();
+
+            Value* list = tryCreateList(
+                formal_type,
+                graph,
+                loc,
+                at::ArrayRef<NamedValue>(args).slice(used_args),
+                failure_messages,
+                err,
+                allow_conversions,
+                type_env);
+            if (!list) {
+              return c10::nullopt;
+            }
+            used_args = args.size();
+            positional_inputs.push_back(list);
+            continue;
           }
-          used_args = args.size();
-          positional_inputs.push_back(list);
-          continue;
         }
-      }
 
-      // Set actual_named_value to the argument and mark the arg position as
-      // used
-      actual_named_value = args[used_args];
-      used_args++;
-    } else if (auto kwarg_idx = findInputWithName(arg.name(), kwargs)) {
-      const NamedValue& nv = kwargs[*kwarg_idx];
-      if (used_kwarg[*kwarg_idx]) {
+        // Set actual_named_value to the argument and mark the arg position as
+        // used
+        actual_named_value = args[used_args];
+        used_args++;
+      } else if (auto kwarg_idx = findInputWithName(arg.name(), kwargs)) {
+        const NamedValue& nv = kwargs[*kwarg_idx];
+        if (used_kwarg[*kwarg_idx]) {
+          if (failure_messages) {
+            err() << "Argument " << nv.name()
+                  << " specified twice in schema, submit a bug report!\n";
+          }
+          return c10::nullopt;
+        }
+        used_kwarg[*kwarg_idx] = true;
+        actual_named_value = nv;
+      } else if (arg.default_value()) {
+        // Argument has a default value and no value was provided, so use the
+        // default
+        actual_named_value = NamedValue(*arg.default_value());
+      } else {
         if (failure_messages) {
-          err() << "Argument " << nv.name()
-                << " specified twice in schema, submit a bug report!\n";
+          err() << "Argument " << schema.arguments()[schema_i].name()
+                << " not provided.\n";
```

Comment:
May be same here too?

Reply:
Not all NamedValues have names, but all Arguments have names. This is a safe alternative that won't throw